clients/client-shell/apis/provider.go

```diff
@@ -228,6 +228,16 @@ func execute(
 	}
 
 	res, err := req.Send()
+	if err != nil {
+		// Got considers redirects an error, but we would like to read the body as if it
```

Comment:
It started in JS with [sindresorhus](https://github.com/sindresorhus/got), but has been ported to go by [TC folks](https://github.com/taskcluster/go-got).  It's not totally clear that was a good idea.  Between this issue, and the similar issue with httpbackoff (which is also a separate library owned by TC), it might make sense to reconsider our choices.  In another PR :)